engine/table/src/main/java/io/deephaven/engine/table/impl/util/JobScheduler.java

```diff
@@ -164,42 +167,45 @@ private void startTasks(
                 scheduler.submit(executionContext, taskInvoker::execute, description,
                         IterationManager::onUnexpectedJobError);
             }
+            decrementReferenceCount();
```

Comment:
This doesn't look right. It's OK to increment the ref count when the task is scheduled but we need to wait until the task completes / throws before decrementing.

Reply:
At least use a try-finally.